File: src/main/java/de/tum/cit/aet/artemis/web/rest/dto/ExerciseManagementStatisticsDTO.java
package de.tum.cit.aet.artemis.web.rest.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record ExerciseManagementStatisticsDTO(double averageScoreOfExercise, double maxPointsOfExercise, int[] scoreDistribution, int numberOfExerciseScores,
        long numberOfParticipations, long numberOfStudentsOrTeamsInCourse, long numberOfPosts, long numberOfResolvedPosts) {
}

File: src/main/webapp/app/exercise/statistics/exercise-management-statistics-dto.ts
export interface ExerciseManagementStatisticsDto {
    // average score
    averageScoreOfExercise: number;
    maxPointsOfExercise: number;
    scoreDistribution: number[];
    numberOfExerciseScores: number;

    // participation rate
    numberOfParticipations: number;
    numberOfStudentsOrTeamsInCourse: number;

    // posts
    numberOfPosts: number;
    numberOfResolvedPosts: number;

    // helper
    absoluteAveragePoints?: number;
    participationsInPercent?: number;
    resolvedPostsInPercent?: number;
}
